module/zfs/zvol.c

```diff
@@ -847,8 +850,10 @@ zvol_discard(void *arg)
 		size = end - start;
 	}
 
-	if (start >= end)
-		goto out;
+	if (start >= end) {
+		error = SET_ERROR(EIO);
```

Comment:
I have no idea what this check is for, we already check for `size == 0` is zvol_request, so I think it is safe to remove this.